acme/acme/crypto_util.py

```diff
@@ -0,0 +1,146 @@
+"""Crypto utilities."""
+import contextlib
+import logging
+import socket
+import sys
+
+from six.moves import range  # pylint: disable=import-error,redefined-builtin
+
+import OpenSSL
+
+from acme import errors
+
+
+logger = logging.getLogger(__name__)
+
+
+_DEFAULT_SSL_METHOD = OpenSSL.SSL.SSLv23_METHOD
+
+
+def _serve_sni(certs, sock, reuseaddr=True, method=_DEFAULT_SSL_METHOD,
+               accept=None):
```

Comment:
This feels out of place to me. This is (basically) what standalone does. If this meant to be an improvement on standalone, why not put the code there in another PR? Why does the acme module need to be able to start a SNI-enabled server?